lib/datadog/kit/appsec/events.rb

```diff
@@ -57,9 +77,13 @@ def self.track_login_failure(trace, user_id:, user_exists:, **others)
         # @param others [Hash<Symbol, String>] Additional free-form
         #   event information to attach to the trace. Key must not
         #   be :track.
-        def self.track(event, trace, **others)
-          trace.set_tag("appsec.events.#{event}.track", 'true')
+        def track(event, trace, **others, &custom_track_tags_block)
+          custom_track_tags_block ||= proc do |trace_op, event_name|
+            trace_op.set_tag("appsec.events.#{event_name}.sdk", 'true')
+          end
 
+          trace.set_tag("appsec.events.#{event}.track", 'true')
+          custom_track_tags_block.call(trace, event)
```

Comment:
It seems to me this API is now very abstraction-leaking. It looks like we're attempting to shapeshifting `track` into something it was not intended to do.

Were it internal I would not be too worried as we could refactor it at our own discretion, but this is veeeery public API so is bound to be used and abused.